### DISCUSSION TITLE
mentoring p4da

### DISCUSSION THREAD

**Discussion by tatapinandhika** [teamalgoritma/community#16] [Q&A] [answered]
bagaimana cara mengetahui data yang paling sering muncul? contohnya data penjualan sebuah ecommerce, bagaimana cara kita tahu item apa yang paling sering terjual?. dan satu lagi pertanyaannya, jika data date apakah typenya sebagai category atau datetime64?, lalu apakah jika format datenya dd/mm/yy bisa di subsetting berdasarkan tahunnya?. makasi
![Untitled](https://user-images.githubusercontent.com/101564223/158958744-7dabd08d-9cc6-4d96-8ae6-ed624e8bca44.png)

**Comment by inytss** [comment]
1. Untuk mengetahui data yang sering muncul terutama pada suatu kolom yang memiliki tipe object dapat menggunakan `df.describe(include = 'object')`. Sebagai contoh disini saya lampirkan output dari deskripsi data `rice`.

![Screenshot from 2022-03-18 15-00-34](https://user-images.githubusercontent.com/49224928/158960454-caf25f9f-e70d-4696-a10e-ee7dab923d8e.png)
Informasi yang menunjukkan nilai yang sering muncul ditunjukkan pada nilai **top**. 

2. Apabila terdapat data berupa date maka akan lebih baik disimpan dalam tipe data datetime64 dibandingkan dalam bentuk category. Untuk format datetime secara default di python akan memiliki format yy/mm/dd, jadi apabila formatnya masih dalam bentuk dd/mm/yy biasanya belum tersimpan dalam tipe data datetime. Bapak dapat menyimpannya dalam tipe data datetime terlebih dahulu dan dapat di ekstrak nilai tahunnya menggunakan asesor dari datetime.

**Comment by dwiknrd** [comment, marked as answer]
1. Untuk mengetahui jumlah kemunculan data, bapak dapat membuat frekuensi tabel. Frekuensi tabel dapat dilakukan dengan menggunakan method `value_counts()` atau `pd.crosstab`.  Berikut adalah dokumentasi dan cara penggunaannya https://pandas.pydata.org/docs/reference/api/pandas.Series.value_counts.html | https://pandas.pydata.org/docs/reference/api/pandas.crosstab.html
Setelah mendapatkan frekuensi tabel bapak dapat mensubset dataframe sesuai dengan frekuensi yang paling banyak atau dapat menggunakan method `max()`

2. Salah satu alasan untuk mengubah tipe data datetime adalah supaya kita dapat mengekstrak informasi dari data tersebut seperti tahun. Bapak dapat membuat kolom baru dengan mengambil informasi tahun dengan cara

```
df['year'] = df['InvoiceDate'].dt.year
```
Setelah itu bapak dapat mensubset bedasarkan tahun yang dinginkan.
